var/spack/repos/builtin/packages/muscle/package.py

```diff
@@ -14,13 +14,23 @@ class Muscle(MakefilePackage):
     homepage = "http://drive5.com/muscle/"
     url      = "http://www.drive5.com/muscle/muscle_src_3.8.1551.tar.gz"
 
+    version('3.8.31', sha256='43c5966a82133bd7da5921e8142f2f592c2b5f53d802f0527a2801783af809ad',
+            url='https://drive5.com/muscle/downloads3.8.31/muscle3.8.31_src.tar.gz')
```

Comment:
You can add a url_for_version to handle the URL change. Also, update the url above do spack versions and checksum work.